python/pyiceberg/catalog/rest.py

```diff
@@ -0,0 +1,373 @@
+from pyiceberg.utils.iceberg_base_model import IcebergBaseModel
+
+
+class Endpoints:
+    get_config: str = "config"
+    list_namespaces: str = "namespaces"
+    create_namespace: str = "namespaces"
+    load_namespace_metadata: str = "namespaces/{namespace}"
+    drop_namespace: str = "namespaces/{namespace}"
+    update_properties: str = "namespaces/{namespace}/properties"
+    list_tables: str = "namespaces/{namespace}/tables"
+    create_table: str = "namespaces/{namespace}/tables"
+    load_table: str = "namespaces/{namespace}/tables/{table}"
+    update_table: str = "namespaces/{namespace}/tables/{table}"
+    drop_table: str = "namespaces/{namespace}/tables/{table}?purgeRequested={purge}"
+    table_exists: str = "namespaces/{namespace}/tables/{table}"
+    get_token: str = "oauth/tokens"
+    rename_table: str = "tables/rename"
+
+
+AUTHORIZATION_HEADER = "Authorization"
+BEARER_PREFIX = "Bearer"
+CATALOG_SCOPE = "catalog"
+CLIENT_ID = "client_id"
+PREFIX = "prefix"
+CLIENT_SECRET = "client_secret"
+CLIENT_CREDENTIALS = "client_credentials"
+CREDENTIAL = "credential"
+GRANT_TYPE = "grant_type"
+SCOPE = "scope"
+TOKEN_EXCHANGE = "urn:ietf:params:oauth:grant-type:token-exchange"
+
+NAMESPACE_SEPARATOR = b"\x1F".decode("UTF-8")
+
+
+class RestTable(Table):
+    config: Properties = Field(default_factory=dict)
+
+
+class CreateTableRequest(IcebergBaseModel):
+    name: str = Field()
+    location: Optional[str] = Field()
+    table_schema: Schema = Field(alias="schema")
+    partition_spec: Optional[PartitionSpec] = Field(alias="partition-spec")
+    write_order: Optional[SortOrder] = Field(alias="write-order")
+    stage_create: bool = Field(alias="stage-create", default=False)
+    properties: Properties = Field(default_factory=dict)
+
+
+class TokenResponse(IcebergBaseModel):
+    access_token: str = Field()
+    token_type: str = Field()
+    expires_in: int = Field()
+    issued_token_type: str = Field()
+
+
+class ConfigResponse(IcebergBaseModel):
+    defaults: Properties = Field()
+    overrides: Properties = Field()
+
+
+class ListNamespaceResponse(IcebergBaseModel):
+    namespaces: List[Identifier] = Field()
+
+
+class NamespaceResponse(IcebergBaseModel):
+    namespace: Identifier = Field()
+    properties: Properties = Field()
+
+
+class UpdateNamespacePropertiesResponse(IcebergBaseModel):
+    removed: List[str] = Field()
+    updated: List[str] = Field()
+    missing: List[str] = Field()
+
+
+class ListTableResponseEntry(IcebergBaseModel):
+    name: str = Field()
+    namespace: Identifier = Field()
+
+
+class ListTablesResponse(IcebergBaseModel):
+    identifiers: List[ListTableResponseEntry] = Field()
+
+
+class ErrorResponseMessage(IcebergBaseModel):
+    message: str = Field()
+    type: str = Field()
+    code: int = Field()
+
+
+class ErrorResponse(IcebergBaseModel):
+    error: ErrorResponseMessage = Field()
+
+
+class RestCatalog(Catalog):
+    token: TokenResponse
+    config: ConfigResponse
```

Comment:
I think this should be tracked a string map, not a `ConfigResponse`. See my comment in `url`

Reply:
Changed this into `Properties`